Desktop apps (#3)
dobromir-hristov/vuecommunity: docs/guide/ecosystem/desktop-apps.md

### Useful resources

- [Vue.js & Electron: The easy way](https://medium.com/@mikeal/vue-js-electron-the-easy-way-adc3ca09234a)
- [Building a Desktop App with Vue: Electron](https://dev.to/vuevixens/building-a-desktop-app-with-vue-electron-3pl)

## NW.js

[NW.js](https://nwjs.io/) (previosly known as node-webkit) is a framework for building desktop applications with HTML, CSS, and JavaScript. It's based on Chromium and Node.js as well as [Electron](#electron). NW.js lets you call Node.js code and modules directly from browser and also use Web technologies in your app. Further, you can easily package a web application to a native application.

### Differences from Electron

Electron and NW.js share many features: they are both built on top of Chromium and Node.js and use HTML, CSS and JS for building a desktop app. But they also have some notable differences:

- In NW.js the main entry point of an application is a web page or a JS script. You specify this entry via `package.json` `main` property. In Electron, the entry point is a JavaScript script. Instead of providing a URL directly, you manually create a browser window and load an HTML file using the API.

> - On build process NW.js builds all Chromium; Electron uses [libchromiumcontent](https://github.com/electron/libchromiumcontent) to access Chromium's Content API

Reviewer: ```suggestion
- During the build process NW.js builds the whole Chromium; Electron uses [libchromiumcontent](https://github.com/electron/libchromiumcontent) to access Chromium's Content API
```

**Pros**

- easy to start with
- no need to change an existing source code
- compiling to protected binaries
- has [Vue CLI plugin](https://github.com/NataliaTepluhina/vue-cli-plugin-nwjs)

**Cons**

- big package size
- high memory usage
- significantly less used than Electron

### Useful resources

- [vue-desktop-basic boilerplate](https://github.com/TheJaredWilcurt/vue-desktop-basic)
- [Building a Desktop App with Vue: NW.js](https://dev.to/vuevixens/building-a-desktop-app-with-vue-nwjs-1f9e)
- [Reducing app distribution size in NW.js](https://dev.to/thejaredwilcurt/reducing-app-distribution-size-in-nwjs-3d5f)

## Vuido

[Vuido](https://vuido.mimec.org/) is a framework for creating native desktop applications based on Vue.js. Applications using Vuido can run on Windows, OS X and Linux, using native GUI components.

Under the hood, Vuido uses the [libui](https://github.com/andlabs/libui) library which provides native GUI components for each desktop platform, and the [libui-node](https://github.com/parro-it/libui-node) bindings for Node.js.

> The core difference of using Vuido comparing to Electron and NW.js is you're not going to use HTML tags or CSS styles: you have a set of UI components and you're limited to it.

Reviewer: ```suggestion
The core difference between Vuido and Electron or NW.js is that you're not going to use HTML tags or CSS styles with Vuido, you are limited to a predefined set of UI components, that the OS platforms provide.
```
